### DISCUSSION TITLE
Exclude Digital signatures

### DISCUSSION THREAD

**Discussion by m4th** [Hakky54/ayza#415] [General]
I have a personal digital signature installed on my pc truststore .
And when i run a simple test with a jdk17 :

```      
        @Test
	void sslF() throws IOException, KeyStoreException, CertificateException, NoSuchAlgorithmException {
		SSLFactory.builder()
				.withSystemTrustMaterial()
				.build();

	}
```

i get a popup from the digital signature software to enter my digital signature password .

I dont know if its the right way to do it but ,
Do you know if there is a way to filter certificates on the keystore to exclude digital signatures ? 
and withSystemTrustMaterial() should impliment this behavior ?

because the problem happens with sonarlint intellij plugin and robot 3T, each time intellij/or robot3t  start the login popup appears ... 

Best regards

**Comment by Hakky54** [comment]
Can you provide an example how you create and add the digital signature?
In that way I can reproduce it on my side and work on a solution. Seems like an annoying issue for you that it asks you the password every time. @m4th can you explain how to add the digital signature in your system truststore? And which OS are you using?

**Comment by m4th** [comment]
Thank for your answer .

I'll try to give you a proper testing case . but i dont have a lot of time to debug more . (maybe this week end..)

Our digital signature has proprietary format due to the cryptographic service provider we are using for the login.

So ill try to export a signature to an open format   ..

I'm not sure but the only diffirence between a digital signature and an other certificat is the keyusage (https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/cert/X509Certificate.html#getKeyUsage())

we reproduce the problem with a simplier test ...

```
      @Test
	void sslF() throws IOException, KeyStoreException, CertificateException, NoSuchAlgorithmException {
			KeyStore keyStore = KeyStore.getInstance("Windows-MY");
		        keyStore.load(null, null);
	}
```

BR

**Reply by Hakky54** [reply]
@m4th 
Did you manage to reproduce creating the digital signature? I tried it on my side with the specific keyusages however it did not work. I was able to create a digital signature and password protect it, however when I use the system trusted certificates I am not prompted to enter my password. So I am still waiting for your input so I can reproduce it on my side.
